src/sage/geometry/hyperplane_arrangement/ordered_arrangement.py

```diff
@@ -0,0 +1,652 @@
+            sage: H = A3(hyperplane_arrangements.braid(4).essentialization())
+            sage: G3 = H.projective_fundamental_group(); G3.sorted_presentation()
+            Finitely presented group
+            < x0, x1, x2, x3, x4 | x4^-1*x3^-1*x2^-1*x3*x4*x0*x2*x0^-1,
+                                   x4^-1*x2^-1*x4*x2, x4^-1*x1^-1*x0^-1*x1*x4*x0,
+                                   x4^-1*x1^-1*x0^-1*x4*x0*x1,
+                                   x4^-1*x1^-1*x3*x0*x1*x3^-1*x2^-1*x4*x0^-1*x2,
+                                   x3^-1*x2^-1*x1^-1*x0^-1*x3*x0*x1*x2,
+                                   x3^-1*x1^-1*x3*x1 >
+            sage: G3.abelian_invariants()
+            (0, 0, 0, 0, 0)
+            sage: A4.<t1, t2, t3, t4> = OrderedHyperplaneArrangements(QQ)
+            sage: H = A4(hyperplane_arrangements.braid(4))
+            sage: G4 = H.projective_fundamental_group(); G4.sorted_presentation()
+            Finitely presented group
+            < x0, x1, x2, x3, x4 | x4^-1*x3^-1*x2^-1*x3*x4*x0*x2*x0^-1,
+                                   x4^-1*x2^-1*x4*x2, x4^-1*x1^-1*x0^-1*x1*x4*x0,
+                                   x4^-1*x1^-1*x0^-1*x4*x0*x1,
+                                   x4^-1*x1^-1*x3*x0*x1*x3^-1*x2^-1*x4*x0^-1*x2,
+                                   x3^-1*x2^-1*x1^-1*x0^-1*x3*x0*x1*x2,
+                                   x3^-1*x1^-1*x3*x1 >
+            sage: G4.abelian_invariants()
+            (0, 0, 0, 0, 0)
+
+            sage: # needs sirocco
+            sage: L.<t0, t1, t2, t3, t4> = OrderedHyperplaneArrangements(QQ)
+            sage: H = hyperplane_arrangements.coordinate(5)
+            sage: H = L(H)
+            sage: g = H.projective_fundamental_group()
+            sage: g.is_abelian(), g.abelian_invariants()
+            (True, (0, 0, 0, 0))
+            sage: L(t0, t1, t2, t3, t4, t0 - 1).projective_fundamental_group()
+            Traceback (most recent call last):
+            ...
+            TypeError: the arrangement is not projective
+            sage: T.<t> = QQ[]
+            sage: K.<a> = NumberField(t^3 + t + 1)
+            sage: L.<x, y, z> = OrderedHyperplaneArrangements(K)
+            sage: H = L(a*x + y - z, x + a*y + z, x - z, y - z)
+            sage: H.projective_fundamental_group()
+            Traceback (most recent call last):
+            ...
+            TypeError: the base field is not in QQbar
+            sage: A.<x> = OrderedHyperplaneArrangements(QQ)
+            sage: H = A(); H
+            Empty hyperplane arrangement of dimension 1
+            sage: H.projective_fundamental_group()
+            Finitely presented group <  |  >
+        """
+        K = self.base_ring()
+        if not K.is_subring(QQbar):
+            raise TypeError('the base field is not in QQbar')
+        if not self.is_linear():
+            raise TypeError('the arrangement is not projective')
+        if self._projective_fundamental_group:
+            return self._projective_fundamental_group
```

Comment:
It is important for this computations that the results may be saved along sessions. One solution was to use `@cached_method(do_pickle=True)` but f.p. groups still have reduction problems, maybe this can be changed if #37128 works.